app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -1,17 +1,104 @@
-// package ch.epfl.favo.common;
-//
-// import com.google.firebase.firestore.FirebaseFirestore;
-//
-// public final class DatabaseWrapper {
-//
-//    private static FirebaseFirestore mFirestore;
-//
-//    public DatabaseWrapper() {
-//        FirebaseFirestore.setLoggingEnabled(true);
-//        mFirestore = FirebaseFirestore.getInstance();
-//    }
-//
-//    public static FirebaseFirestore getDatabaseInstance() {
-//        return mFirestore;
-//    }
-// }
+package ch.epfl.favo.common;
+
+import android.util.Log;
+
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class DatabaseWrapper<T extends DatabaseDocument> {
```

Comment:
It as a good thing that you have modularized the database into its own abstraction :+1: